### PR TITLE
fix(AppHeaderCatalogDropdown): handle data science paths better

### PR DESCRIPTION
**Pull request by borisonr**
### Overview

instead of having two separate links for data science paths, we'll link to the data science specialization landing page, which makes the styles and ux clearer

Screen Shot 2022-06-21 at 4 35 29 PM

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [DISC-697](https://codecademy.atlassian.net/browse/DISC-697)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

[portal app integration pr](https://github.com/codecademy-engineering/portal-app/pull/2125)
[portal app integration preview](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2125)
[monolith pr](https://github.com/codecademy-engineering/Codecademy/pull/31159)
[monolith preview](https://pr-31159-monolith.dev-eks.codecademy.com/)

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.146-alpha.f9bb66.0<br/>@codecademy/gamut-labs@32.1.1-alpha.f9bb66.0<br/>@codecademy/styleguide@58.0.9-alpha.f9bb66.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b206fe6e8f7a0da1e9d926--gamut-preview.netlify.app>https://62b206fe6e8f7a0da1e9d926--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b206fe6e8f7a0da1e9d926>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b20a2695ad320e46169960--gamut-preview.netlify.app>https://62b20a2695ad320e46169960--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b20a2695ad320e46169960>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.146-alpha.2abc2b.0<br/>@codecademy/gamut-labs@32.1.1-alpha.2abc2b.0<br/>@codecademy/styleguide@58.0.9-alpha.2abc2b.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.146-alpha.065cb1.0<br/>@codecademy/gamut-labs@32.1.1-alpha.065cb1.0<br/>@codecademy/styleguide@58.0.9-alpha.065cb1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b216c7f0c5b31b5b3a30ef--gamut-preview.netlify.app>https://62b216c7f0c5b31b5b3a30ef--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b216c7f0c5b31b5b3a30ef>Deploy Logs</a></p>

**Comment by aresnik11** [comment]
@borisonr hey! how do i trigger the new catalog menu? thanks!

**Comment by aresnik11** [comment]
> > @borisonr hey! how do i trigger the new catalog menu? thanks!
> 
> hmm you dont automatically see it? i have the variant on to 100% in non-prod envs @aresnik11

ah i see it if i'm logged in

**Comment by borisonr** [comment]
> > > @borisonr hey! how do i trigger the new catalog menu? thanks!
> > 
> > 
> > hmm you dont automatically see it? i have the variant on to 100% in non-prod envs @aresnik11
> 
> ah i see it if i'm logged in

oh you might have been cached into the control before i turned the variant up to 100%

**Review by zippyzow — APPROVED** [review]
These changes look good on my end

**Review by aresnik11 — CHANGES_REQUESTED** [review]
I'm unable to click on any of the links in the dropdown. They look like links and I can see that it is an anchor element, but clicking on it doesn't do anything.

I was also able to open the resources menu on top of the catalog menu a few times, still trying to figure out how to consistently replicate that though
Screen Shot 2022-06-22 at 11 19 16 AM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.148-alpha.2ebee4.0<br/>@codecademy/gamut-labs@32.2.1-alpha.2ebee4.0<br/>@codecademy/styleguide@58.1.1-alpha.2ebee4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b34025c6dc3a203e5a58fd--gamut-preview.netlify.app>https://62b34025c6dc3a203e5a58fd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b34025c6dc3a203e5a58fd>Deploy Logs</a></p>

**Comment by zippyzow** [comment]
> this is so weird, i'm kind of stumped... it works if you make the screen smaller so that it's the hamburger menu... any ideas what may be happening @zippyzow or @aresnik11 ? (re links not being clickable, i cant reproduce the overlap with the resources dropdown) also interestingly it works if you tab/arrow into it

Taking a look on my branch, I'll let you know if I figure it out

**Comment by aresnik11** [comment]
> this is so weird, i'm kind of stumped... it works if you make the screen smaller so that it's the hamburger menu... any ideas what may be happening @zippyzow or @aresnik11 ? (re links not being clickable, i cant reproduce the overlap with the resources dropdown) also interestingly it works if you tab/arrow into it

also looking! i've confirmed what you said re links work in hamburger menu but not in regular dropdown, and they work if you access via keyboard and click enter

**Comment by zippyzow** [comment]
@borisonr I placed the anchor directly inside of the `StyledAnchorColumn` and it worked (like screenshot below). Seems like there needs to be a separate `onClick` handler to trigger the `onClick` handler in the `CatalogLink` component or something. 

Screen Shot 2022-06-22 at 2 41 23 PM

**Comment by borisonr** [comment]
> @borisonr I placed the anchor directly inside of the `StyledAnchorColumn` and it worked (like screenshot below). Seems like there needs to be a separate `onClick` handler to trigger the `onClick` handler in the `CatalogLink` component or something.
> 
> Screen Shot 2022-06-22 at 2 41 23 PM

ahh amazing find, thank you @zippyzow !!

File: packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx [hunk 1]
```diff
@@ -158,17 +129,18 @@ export const AppHeaderCatalogSection = React.forwardRef<
...
-                    key={item.id}
-                    size={{ _: 4 }}
-                    my="auto"
-                    minHeight={46}
-                  >
-                    <StyledAnchorBox width="12rem">
-                      <CatalogLink item={item} />
-                    </StyledAnchorBox>
+                  <Column key={item.id} size={{ _: 4 }}>
+                    <Anchor
```
Comment (borisonr) [inline packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx:135]: there was a bunch of excess css i was able to remove so there became less of a need to set up the cataloglink component anyway but lemme know if you want me to keep that and properly scope the click handler

[the remark this answers is not in the document]
Comment (zippyzow) [reply packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx:135]: This is good with me

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.148-alpha.7f09cd.0<br/>@codecademy/gamut-labs@32.2.1-alpha.7f09cd.0<br/>@codecademy/styleguide@58.1.1-alpha.7f09cd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b37051baf8fd1f6cbd77f0--gamut-preview.netlify.app>https://62b37051baf8fd1f6cbd77f0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b37051baf8fd1f6cbd77f0>Deploy Logs</a></p>

**Review by aresnik11 — COMMENTED** [review]
The "explore full catalog" link still isn't clickable but the rest of them work as expected! Also noticed that "Cybersecurity" and "Machine Learning" aren't accessible via keyboard. Is the capitalization on "Top career Paths" correct? seems off to me, but I'll defer to you/your designer.

Re the double menu issue, I have replication steps now but I can replicate on 2 dropdowns in the app header outside of the mega menu so I'm going to make a bug ticket for web plat

**Comment by borisonr** [comment]
> The "explore full catalog" link still isn't clickable but the rest of them work as expected! Also noticed that "Cybersecurity" and "Machine Learning" aren't accessible via keyboard. Is the capitalization on "Top career Paths" correct? seems off to me, but I'll defer to you/your designer.
> 
> Re the double menu issue, I have replication steps now but I can replicate on 2 dropdowns in the app header outside of the mega menu so I'm going to make a bug ticket for web plat

thank you for all the thorough QA @aresnik11 ! just pushed those fixes and will rerequest after i update the preview envs

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62b46ba93fcb4523bdd03248--gamut-preview.netlify.app>https://62b46ba93fcb4523bdd03248--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62b46ba93fcb4523bdd03248>Deploy Logs</a></p>

**Pull request merged**